Source/JavaScriptCore/wasm/WasmTypeDefinition.cpp

```diff
@@ -98,6 +132,20 @@ void StructType::dump(PrintStream& out) const
     out.print(")");
 }
 
+StructType::StructType(FieldType* payload, StructFieldCount fieldCount, const FieldType* fieldTypes)
+    : m_payload(payload)
+    , m_fieldCount(fieldCount)
+{
+    Checked<unsigned> currentFieldOffset = 0;
+    for (unsigned fieldIndex = 0; fieldIndex < m_fieldCount; ++fieldIndex) {
+        getField(fieldIndex) = fieldTypes[fieldIndex];
+        *getFieldOffset(fieldIndex) = currentFieldOffset.value();
+        currentFieldOffset += typeKindSizeInBytes(field(fieldIndex).type.kind);
+    }
+
+    m_instancePayloadSize = WTF::roundUpToMultipleOf<sizeof(uint64_t)>(currentFieldOffset.value());
```

Comment:
This is not what we would like to have. We should fail the parsing, not crashing.

Reply:
We need to ensure that,
1. FunctionParser checks this overflow and fail the parsing
2. We should have a test for that.